controllers/imageupdateautomation_controller.go

```diff
@@ -261,22 +261,33 @@ func (r *ImageUpdateAutomationReconciler) Reconcile(ctx context.Context, req ctr
 
 	switch {
 	case auto.Spec.Update != nil && auto.Spec.Update.Strategy == imagev1.UpdateStrategySetters:
-		// For setters we first want to compile a list of _all_ the
-		// policies in the same namespace (maybe in the future this
-		// could be filtered by the automation object).
-		var policies imagev1_reflect.ImagePolicyList
-		if err := r.List(ctx, &policies, &client.ListOptions{Namespace: req.NamespacedName.Namespace}); err != nil {
+		withoutDiscriminator := func (p imagev1_reflect.ImagePolicy) bool{
+			return p.Spec.FilterTags==nil || p.Spec.FilterTags.Discriminator == ""
```

Comment:
There are formatting inconsistencies here. Please use `gofmt`, running make locally before committing would do.
```suggestion
		withoutDiscriminator := func (p imagev1_reflect.ImagePolicy) bool {
			return p.Spec.FilterTags == nil || p.Spec.FilterTags.Discriminator == ""
```

Reply:
will do.